### PR TITLE
added user roles and did some styling

### PR DESCRIPTION
**Pull request by AbdullahAlger**
Added the user roles and did a little styling. More to do on the styling.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -6,5 +6,16 @@ class User < ActiveRecord::Base
...
+    role == 'admin'
+  end
+
+  def standard
+    role == 'standard'
+  end
+
+  def premium
+    role == 'premium'
+  end
```
Comment (eliotsykes) [inline app/models/user.rb]: Consider being consistent with the naming of these `role?` convenience methods, adding question mark suffixes to all.

File: app/policies/wiki_policy.rb [hunk 2]
```diff
@@ -1,2 +1,22 @@
 class WikiPolicy < ApplicationPolicy
+
+  class Scope < Scope
+    def resolve
+      if user.present? && (user.admin?)
```
Comment (eliotsykes) [inline app/policies/wiki_policy.rb]: Parentheses can be removed from line 5

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -5,23 +5,23 @@
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
   <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
+  <script src="http://use.edgefonts.net/alegreya.js"></script>
```
Comment (eliotsykes) [inline app/views/layouts/application.html.erb]: Using `http:` URLs for assets inside of a page that is served over HTTPS opens up some security vulnerabilities. It also shows users mixed content warning in the browser. Here's how it'd look in Chrome:

mixed content warning

You'd see this warning on your Production environment when visiting the app at the HTTPS URL `https://your-app-name.herokuapp.com`.

Thankfully this font provider supports HTTPS, so you can change the URL on line 8 to be `https://` and that will solve the mixed content warning problem in this case.

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -5,23 +5,23 @@
...
-        Hello, <%= current_user.email %>! <%= link_to "Sign Out", destroy_user_session_path, method: :delete %>
-      <% else %>
-        <%= link_to "Sign In", new_user_session_path %> or
-        <%= link_to "Register", new_user_registration_path %>
-      <% end %>
-    </div>
+    <li><%= link_to "Wikis", wikis_path %></li>
   </ul>
+  <div class="navbar-right user-info">
+    <% if current_user %>
```
Comment (eliotsykes) [inline app/views/layouts/application.html.erb]: Devise also provides a `user_signed_in?` method you can use in views. Consider using it when it helps make it easier to follow the code's purpose. In this case the code would reveal its intention a little more clearly if line 18 became:

```
  <% if user_signed_in? %>
    ... info to show to signed in users only ...
```

File: config/environments/production.rb [hunk 5]
```diff
@@ -78,4 +78,8 @@
   config.active_record.dump_schema_after_migration = false
 
   config.action_mailer.default_url_options = { host: 'miwiki.herokuapp.com'}
+
+
 end
+
+Rails.application.routes.default_url_options[:host] = 'localhost:3000'
```
Comment (eliotsykes) [inline config/environments/production.rb]: This isn't right for the production environment, remove line 85 entirely. Line 80 is serving the purpose you want I believe.

File: config/environments/development.rb [hunk 6]
```diff
@@ -44,4 +44,7 @@
   config.action_mailer.perform_deliveries = true
   config.action_mailer.raise_delivery_errors = true
 
+
 end
+
+Rails.application.routes.default_url_options[:host] = 'localhost:3000'
```
Comment (eliotsykes) [inline config/environments/development.rb]: Line 42 may be enough that this line 50 is not needed. Try removing line 50 and send me any errors you get.

Comment (AbdullahAlger) [inline config/environments/development.rb]: Done. No errors. I think we are working in parallel here. I keep getting your messages!

File: config/environments/test.rb [hunk 7]
```diff
@@ -40,3 +40,5 @@
   # Raises error for missing translations
   # config.action_view.raise_on_missing_translations = true
 end
+
+Rails.application.routes.default_url_options[:host] = 'localhost:3000'
```
Comment (eliotsykes) [inline config/environments/test.rb]: Remove line 44 and add this config at line 33:
```
config.action_mailer.default_url_options = { host: 'localhost:3000' }
```
If this doesn't work please send me the errors.

File: config/environments/production.rb (hunk 5 above)
Comment (AbdullahAlger) [inline config/environments/production.rb]: Done

File: config/environments/test.rb (hunk 7 above)
Comment (AbdullahAlger) [inline config/environments/test.rb]: Working.

**Comment by eliotsykes** [comment]
Ha yep working at the same time :-) hello!

Currently looks like blocipedia doesn't limit controller actions to logged in users. You may want to consider introducing some code like this `before_action`:

https://github.com/AbdullahAlger/bloccit/blob/249d56bef63e18421a25503fb9effd8ed7f05acb/app/controllers/users_controller.rb#L3

The `authenticate_user!` method referenced is provided by Devise. If a user isn't logged in they are prevented from performing the action(s) in the controller.

**Comment by AbdullahAlger** [comment]
Fixed that.

File: app/models/user.rb (hunk 1 above)
Comment (AbdullahAlger) [inline app/models/user.rb]: Done

File: app/views/layouts/application.html.erb (hunk 4 above)
Comment (AbdullahAlger) [inline app/views/layouts/application.html.erb]: Done

File: app/views/layouts/application.html.erb (hunk 3 above)
Comment (AbdullahAlger) [inline app/views/layouts/application.html.erb]: Done. How else could you refer to hosted fonts or libraries.

Comment (eliotsykes) [inline app/views/layouts/application.html.erb]: Different hosts may offer different options. See the hosts own instructions on their options. Hopefully they'll document the trade-offs of each solution so you can make an informed decision about which option you want to use.

Comment (AbdullahAlger) [inline app/views/layouts/application.html.erb]: I've already changed the code after reading some docs.

**Pull request merged**